lagrangebench/defaults.py

```diff
@@ -1,70 +1,189 @@
-        neighbor_list_backend: neighbor list routine. Default "jaxmd_vmap".
-        neighbor_list_multiplier: multiplier for neighbor list capacity. Default 1.25.
-    """
-
-    # training
-    seed: int = 0  # random seed
-    batch_size: int = 1  # batch size
-    step_max: int = 5e5  # max number of training steps
-    dtype: jnp.dtype = jnp.float64  # data type for preprocessing
-    magnitude_features: bool = False  # whether to include velocity magnitude features
-    isotropic_norm: bool = False  #  whether to normalize dimensions equally
-    num_workers: int = 4  # number of workers for data loading
-
-    # learning rate
-    lr_start: float = 1e-4  # initial learning rate
-    lr_final: float = 1e-6  # final learning rate (after exponential decay)
-    lr_decay_steps: int = 1e5  # number of steps to decay learning rate
-    lr_decay_rate: float = 0.1  # learning rate decay rate
-
-    noise_std: float = 3e-4  # standard deviation of the GNS-style noise
-
-    # evaluation
-    input_seq_length: int = 6  # number of input steps
-    n_rollout_steps: int = -1  # number of eval rollout steps. -1 is full rollout
-    eval_n_trajs: int = 1  # number of trajectories to evaluate
-    rollout_dir: str = None  # directory to save rollouts
-    out_type: str = "none"  # type of output. None means no rollout is stored
-    n_extrap_steps: int = 0  # number of extrapolation steps
-    metrics_stride: int = 10  # stride for e_kin and sinkhorn
-    batch_size_infer: int = 2  # batch size for validation/testing
-
-    # logging
-    log_steps: int = 1000  # number of steps between logs
-    eval_steps: int = 10000  # number of steps between evaluations and checkpoints
-
-    # neighbor list
-    neighbor_list_backend: str = "jaxmd_vmap"  # backend for neighbor list computation
-    neighbor_list_multiplier: float = 1.25  # multiplier for neighbor list capacity
+"""Default lagrangebench configs."""
+
+
+from omegaconf import DictConfig, OmegaConf
+
+
+def set_defaults(cfg: DictConfig = OmegaConf.create({})) -> DictConfig:
+    """Set default lagrangebench configs."""
+
+    ### global and hardware-related configs
+    cfg.main = OmegaConf.create({})
+
+    # configuration file. Either main.config or main.model_dir must be specified.
+    # If main.config is specified, main.model_dir is ignored.
+    cfg.main.config = None
+    # Load checkpointed model from this directory
+    cfg.main.model_dir = None
+    # One of "train", "infer" or "all" (= both)
+    cfg.main.mode = "all"
+    # random seed
+    cfg.main.seed = 0
+    # data type for preprocessing. One of "float32" or "float64"
+    cfg.main.dtype = "float64"
+    # gpu device. -1 for CPU. Should be specified before importing the library.
+    cfg.main.gpu = None
+    # XLA memory fraction to be preallocated. The JAX default is 0.75.
+    # Should be specified before importing the library.
+    cfg.main.xla_mem_fraction = None
+    # data directory
+    cfg.main.data_dir = None
```

Comment:
What about not nesting these? Especially `main.config`, so one can just call `python main.py config=configs/...`

Reply:
I chose the name "main" as some of these things are only used in the main.py, i.e. main.config and main.xla_mem_fraction. But we can move all of them out - I indeed didn't see the "main" nesting in other projects.

Reply:
One advantage of cfg.main.* is easier searching for stuff through the code, but this is a minor point

Reply:
ok, I'll remove the .main. hierarchy level